### PR TITLE
GH-32: Initial Setup of the User Service

### PR DESCRIPTION
**Pull request by walidoow**

### CODE REVIEW

**Comment by walidoow** [comment]
For the reviewers, please only take a look at the files under src/, the build.gradle and the docker-compose.yml files because the rest of the files were are all generated automatically 👍

File: Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@EqualsAndHashCode(callSuper = false)
+@SuperBuilder(setterPrefix = "with")
+@Document("user")
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString
+@Data
+public class User extends BaseEntity {
```
Comment (patrickmac3) [inline Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java:17]: would the user class not require a password ?

Comment (walidoow) [reply Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java:17]: The password will be managed by Keycloak and will not be handled by us in the database

Comment (walidoow) [reply Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java:17]: I have not setup Auth yet

**Review by patrickmac3 — COMMENTED** [review]
Everything looks good. I only have a 2 comments/suggestions: 

1. Would a user model not require a password field ? For logging in ? 
2. Rather than having the following packages 
         -  request 
         - response
             - response/user 
    would it not be cleaner in the code base to have a structure that looks more like 
            - dto (or some other namee) 
            - dto/user/ (which includes userRequest and userResponse.

File: Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java (hunk 1 above)
Comment (patrickmac3) [reply Microservices/user-service/src/main/java/app/sportahub/userservice/model/user/User.java:17]: Ok perfect ! Thanks !

**Review by patrickmac3 — APPROVED** [review]
Everything looks good ! Thank you for the changes @walidoow

**Pull request merged**